packages/app-store/office365video/lib/VideoApiAdapter.ts

```diff
@@ -97,8 +98,34 @@ const TeamsVideoApiAdapter = (credential: CredentialForCalendarServiceWithTenant
     },
     invalidateTokenObject: () => oAuthManagerHelper.invalidateCredential(credential.id),
     expireAccessToken: () => oAuthManagerHelper.markTokenAsExpired(credential),
-    updateTokenObject: (tokenObject) => {
-      if (!Boolean(credential.delegatedTo)) {
+    updateTokenObject: async (tokenObject) => {
+      if (!!credential.delegatedTo?.id) {
+        const existingToken = await prisma.delegationCredentialAccesssToken.findFirst({
```

Comment:
We should create repository for DelegationCredentialAccesssToken and use that instead here.